misk/src/main/kotlin/misk/web/jetty/RealWebSocket.kt

```diff
@@ -0,0 +1,98 @@
+package misk.web.jetty
+
+import misk.web.actions.WebSocket
+import misk.web.actions.WebSocketListener
+import okio.ByteString
+import okio.Utf8
+import org.eclipse.jetty.websocket.api.Session
+import org.eclipse.jetty.websocket.api.WebSocketAdapter
+import org.eclipse.jetty.websocket.api.WriteCallback
+import java.util.ArrayDeque
+
+private const val MAX_QUEUE_SIZE = 16 * 1024 * 1024
+
+class RealWebSocket : WebSocket {
+
+  /** Total size of messages enqueued and not yet transmitted by Jetty. */
+  private var outgoingQueueSize = 0L
+
+  /** Messages to send when the Web Socket connects. */
+  private var queue = ArrayDeque<String>()
```

Comment:
later we’ll need like a `Message` abstraction in here so it can be either a String or a ByteString